react/features/transcription/reducer.js

```diff
@@ -0,0 +1,122 @@
+import {
+    ENDPOINT_MESSAGE_RECEIVED,
+    ADD_TRANSCRIPT_MESSAGE,
+    UPDATE_TRANSCRIPT_MESSAGE,
+    REMOVE_TRANSCRIPT_MESSAGE
+} from './actionTypes';
+import { getUpdatedTranscriptionParagraphs } from './functions';
+import { ReducerRegistry } from '../base/redux';
+
+/**
+ * Default State for 'features/transcription' feature
+ */
+const defaultState = {
+    transcriptMessages: {},
+    transcriptionSubtitles: []
+};
+
+/**
+ * Listen for actions for the transcription feature to be used by the actions
+ * to update the rendered transcription subtitles.
+ */
+ReducerRegistry.register('features/transcription', (
+        state = defaultState, action) => {
+
+    switch (action.type) {
+    case ENDPOINT_MESSAGE_RECEIVED:
+        return _endpointMessageReceived(state);
```

Comment:
It seems to me ENDPOINT_MESSAGE_RECEIVED is really a pub/sub event that ends up dispatching ADD/UPDATE/REMOVE. Is it necessary for it to actually go through reducing? If so, could you tell me why because I'm not seeing that part.